rest/context.go

```diff
@@ -325,6 +351,60 @@ func (ctx *gorillaRequestContext) NextURL() (string, error) {
 	return u.String(), nil
 }
 
+func (ctx *gorillaRequestContext) buildURL(fullPath bool, resourceName string,
+	methodName string, pairs ...string) (string, error) {
+	r, ok := ctx.Request()
+	if !ok {
+		return "", fmt.Errorf("unable to build URL for resource name %q: no request available.",
+			resourceName)
+	}
+
+	routeName := resourceName + ":" + methodName
+	route := ctx.router.Get(routeName).Host(r.Host)
+
+	var builder func(pairs ...string) (*url.URL, error)
```

Comment:
Users would have to include the package name, so it would really be `rest.pairs`:

```go
url, err := rest.BuildURL("resource", rest.HandleCreate, rest.pairs{"foo": "bar"})
```

still less off-putting than `map[string]string{}` at least... I'd like to hear what Tyler thinks.